src/components/TodoList/TodoList.jsx

```diff
@@ -1,44 +1,126 @@
 import React from 'react';
 import './TodoList.scss';
+import classNames from 'classnames';
+import PropTypes from 'prop-types';
 
-export const TodoList = () => (
-  <div className="TodoList">
-    <h2>Todos:</h2>
-
-    <div className="TodoList__list-container">
-      <ul className="TodoList__list">
-        <li className="TodoList__item TodoList__item--unchecked">
-          <label>
-            <input type="checkbox" readOnly />
-            <p>delectus aut autem</p>
-          </label>
-
-          <button
-            className="
-              TodoList__user-button
-              TodoList__user-button--selected
-              button
-            "
-            type="button"
-          >
-            User&nbsp;#1
-          </button>
-        </li>
-
-        <li className="TodoList__item TodoList__item--checked">
-          <label>
-            <input type="checkbox" checked readOnly />
-            <p>distinctio vitae autem nihil ut molestias quo</p>
-          </label>
-
-          <button
-            className="TodoList__user-button button"
-            type="button"
+export class TodoList extends React.PureComponent {
+  state = {
+    todos: this.props.todos,
+    searchQuery: '',
+    completedFilter: 'all',
+  }
+
+  handleChange = (event) => {
+    const { value, name } = event.target;
+
+    this.setState({
+      [name]: value,
+    });
+  }
+
+  getVisibleTodos = (todos, searchQuery) => {
+    const normalizedQuery = searchQuery.toLowerCase();
+
+    return todos.filter(todo => todo.title
+      && todo.title.toLowerCase().includes(normalizedQuery));
+  }
+
+  filterCompletedTodos = (todos, completedFilter) => todos.filter(todo => (
+    completedFilter === 'completed'
+      ? todo.completed
+      : !todo.completed
+  ))
```

Comment:
I think in this case switch will better, because if you will want to add one more sorting option it will be difficult